apps/wallet/src/ui/app/components/menu/content/AccountActions.tsx

```diff
@@ -4,24 +4,40 @@
 import { type SuiAddress } from '@mysten/sui.js';
 
 import { useNextMenuUrl } from '../hooks';
+import { AccountType } from '_src/background/keyring/Account';
 import { Link } from '_src/ui/app/shared/Link';
+import { Text } from '_src/ui/app/shared/text';
 
 export type AccountActionsProps = {
     accountAddress: SuiAddress;
+    accountType: AccountType;
 };
 
-export function AccountActions({ accountAddress }: AccountActionsProps) {
+export function AccountActions({
+    accountAddress,
+    accountType,
+}: AccountActionsProps) {
     const exportAccountUrl = useNextMenuUrl(true, `/export/${accountAddress}`);
+    const canExportPrivateKey =
+        accountType === AccountType.DERIVED ||
+        accountType === AccountType.IMPORTED;
+
     return (
         <div className="flex flex-row flex-nowrap items-center flex-1">
-            <div>
-                <Link
-                    text="Export Private Key"
-                    to={exportAccountUrl}
-                    color="heroDark"
-                    weight="medium"
-                />
-            </div>
+            {canExportPrivateKey ? (
+                <div>
+                    <Link
+                        text="Export Private Key"
+                        to={exportAccountUrl}
+                        color="heroDark"
+                        weight="medium"
+                    />
+                </div>
+            ) : (
+                <Text variant="bodySmall" weight="medium" color="steel-dark">
```

Comment:
Should we just not make the row expandable instead?

Reply:
I improvised the design since it was additional work to render the exact same UI but not in a disclosure. I'll work with Abhi on this - I think one reason for keeping the row expandable is for when we add account nicknames